src/accelerate/data_loader.py

```diff
@@ -606,7 +606,16 @@ def _fetch_batches(self, iterator):
                     batches = []
                     for _ in range(self.state.num_processes):
                         batches.append(next(iterator))
-                    batch = concatenate(batches, dim=0)
+                    try:
+                        batch = concatenate(batches, dim=0)
+                    except RuntimeError:
+                        raise RuntimeError(
+                            "You can't use batches of different size with `dispatch_batches=True` or when using an `IterableDataset`."
+                            "As an alternative, you can either: 1) Have batches of same size 2) Set `dispatch_batches=False`. "
+                            "This way, each process will get fetch their own batch. "
+                            "3) Set `split_batches=True`. By doing so, the main process will fetch a full batch and "
+                            "slice it into `num_processes` batches for each process."
```

Comment:
```suggestion
                            "either pass `dispatch_batches=False` and have each process fetch its own batch "
                            " or pass `split_batches=True`. By doing so, the main process will fetch a full batch and "
                            "slice it into `num_processes` batches for each process."
```

I don't think we need 1, since the user has gathered from the error that 1 won't happen if the batch == the batch size.